### DISCUSSION TITLE
Display match occurence on blacklist (address, IP or word)

### DISCUSSION THREAD

**Discussion by adminsyspro** [rspamd/rspamd#3888] [Q&A]
Hello,
I have several rules to block adresses, IP or words, it work's great but it doesn't display the occurence which has triggered the rule.
For example, I have this map : 

BAD_BODY_CUSTOM {
        type = "content";
        filter = "text";
        map = "http://127.0.0.1:8000/mailutils/get_bwl/words/body";
        symbol = "BAD_BODY_CUSTOM"
        regexp = true;
        score = 99.0;
}

When this rule is triggered, I can see in WebUI : 

BAD_BODY_CUSTOM (99), but I dont'see which word.

However, on the same rule but based on subject, It works : 

BAD_SUBJECT_CUSTOM {
        type = "header";
        header = "subject";
        map = "http://127.0.0.1:8000/mailutils/get_bwl/words/subject";
        symbol = "BAD_SUBJECT_CUSTOM"
        regexp = true;
        score = 99.0;
}

In web UI, I can see : 

BAD_SUBJECT_CUSTOM (99) [Congratulations! You can get a $90 American Airlines gift card!]

How can I solve this ?

regards,

**Comment by JustKiddingCode** [comment]
I want the opposite, I want to disable that the match is saved as option to the symbol.
It looks like that's not that easy.
In `src/plugins/lua/multimap.lua` see `value_types`. Content is the only one without an option (`return nil`).

**Reply by adminsyspro** [reply]
Indeed, you're right : 

content = {
    get_value = function() return nil end,
  },

But I think there is a reason to this ?!

I will try to fix that and see if I can get the value in the WebUI.

**Comment by adminsyspro** [comment]
I just tried and it works with this line : 

 content = {
    get_value = function(val) return val end,
  },

For example with a blacklist on word "#Siegesociale:", I got : 

BAD_BODY_CUSTOM	 198	("#Siegesociale:ï¿½ï¿½ï¿½ï¿½ï¿½ï¿½[cid:image001.jpg@01D7A62B.10A17420]ï¿½ï¿½ï¿½ï¿½ï¿½ï¿½,#Siegesociale:ï¿½ï¿½")

But it seems the word is catch two times.

My regexp is like this : 

/\#Siegesociale:\s/i

Do you think it is right ?

**Reply by JustKiddingCode** [reply]
What's the input? Could be an multipart alternativ with html + plain?

**Comment by adminsyspro** [comment]
The input is from a content of mail body in html. In my rule, I set type = "content"; and filter = "text"; 

Maybe I must changed that...

**Comment by adminsyspro** [comment]
I have the same behavior with a configuration in /etc/rspamd/settings.conf : 

settings = "http://127.0.0.1:8000/mailutils/get_bwl/@";

It return for exemple all BL and WL of users like that :

7xnmhybk5ewbblsq {
                from = /^sdgfdsfg\.dsfgdfsggdf@dsfgsdfgsdfg\.com$/i;
                mime_from = /^sdgfdsfg\.dsfgdfsggdf@dsfgsdfgsdfg\.com$/i;
                priority = high;
                rcpt = /^dsfsqdf@sdqfsqdfq\.org$/i;
                mime_rcpt = /^dsfsqdf@sdqfsqdfq\.org$/i;
                apply "default" {
                            CUSTOM_WL_USER = -999.0;
                            action = "no action";
                            greylist = null;
                            }
                            symbols [
                            CUSTOM_WL_USER
                            ]
                }

All works fin, but it's the same, It's doesn't display the occurence which has matched.

I think I must check in the file /usr/share/rspamd/plugins/settings.lua but it seem more tricky.

**Comment by adminsyspro** [comment]
Nobody has an idea ?

Regards,
